internal/drivers/filesystem/storage.go

```diff
@@ -0,0 +1,267 @@
+/******************************************************************************
+*
+*  Copyright 2022 ruilopes.com
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+******************************************************************************/
+
+// TODO the error returned here are displayed to the user as-is. is that a problem? like, leaking path names?
+// TODO wrap all the errors?
+
+package filesystem
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sapcc/keppel/internal/keppel"
+	"github.com/spf13/afero"
```

Comment:
I generally want to avoid additional library dependencies if the stdlib can serve us instead. I understand the value of the abstractions that afero provides, but in this case we're only really using it to prepend `rootPath` to all paths, and that's something that we could just be doing ourselves in `func blobBasePath` etc. instead.

Reply:
Using it also makes it possible to run unit tests from a memory filesystem and maybe replacing the trivial driver with this driver.

If that's not relevant at the moment, let me known, and I can replace it with stdlib.

Reply:
Yeah, we aren't going to need that. There is already an in-memory filesystem in the kernel with tmpfs, we don't need to carry code around to simulate it.

Reply:
Its now using the stdlib `os` package.